backend_new/src/services/application.service.ts

```diff
@@ -559,6 +586,28 @@ export class ApplicationService {
     });
   }
 
+  async authorizeAction(
+    user: User,
+    applicaton: Application,
+    listingId: string,
+    action: permissionActions,
+  ): Promise<void> {
+    const listingJurisdiction = await this.prisma.jurisdictions.findFirst({
+      where: {
+        listings: {
+          some: {
+            id: listingId,
+          },
+        },
+      },
+    });
+    await this.permissionService.canOrThrow(user, 'application', action, {
+      ...applicaton,
```

Comment:
not quite, for applications we use a few different parts depending on what type of user is attempting the action.

For jurisdictional admin we test that the jurisdiction id associated with the application is valid for the particular juris admin user

For partner users we test if the listing the application is tied to is valid for the particular partner user

This can be seen in `addUserPermissions()` of the ` permission.service.ts` file hopefully that helps!

Reply:
I guess I'm still lost on where this is being used. I don't see the `obj` variable being passed into the `addUserPermissions` function and the only reference I see use jurisdictionId, id, listingId, and userId`

Reply:
updated!